File: dotnet/Library/Okapi/DIDKey.cs
using Okapi.Keys.V1;

namespace Okapi.Keys;

public static class DIDKey
{
    /// <summary>
    ///     Generate new key
    /// </summary>
    /// <param name="request"></param>
    /// <returns></returns>
    public static GenerateKeyResponse Generate(GenerateKeyRequest request)
    {
        return Native.Call<GenerateKeyRequest, GenerateKeyResponse>(request, Native.didkey_generate);
    }

    public static ResolveResponse Resolve(ResolveRequest request)
    {
        return Native.Call<ResolveRequest, ResolveResponse>(request, Native.didkey_resolve);
    }
}

File: python/trinsicokapi/didkey.py
from trinsicokapi.proto.okapi.keys.v1 import GenerateKeyRequest, GenerateKeyResponse, ResolveResponse, ResolveRequest
from trinsicokapi.wrapper import _typed_wrap_and_call


def generate(request: GenerateKeyRequest) -> GenerateKeyResponse:
    response = _typed_wrap_and_call("didkey_generate", request, GenerateKeyResponse)
    return response


def resolve(request: ResolveRequest) -> ResolveResponse:
    response = _typed_wrap_and_call("didkey_resolve", request, ResolveResponse)
    return response
